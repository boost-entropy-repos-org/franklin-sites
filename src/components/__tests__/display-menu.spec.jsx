import React from 'react';
import { render, fireEvent, waitForElement } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import DisplayMenu from '../display-menu';
import displayMenuData from '../__mocks__/displayMenu';

let rendered;
const content1 = 'Page 1 main content';
const content2 = 'Page 2 main content';
const itemContent1 = 'Something';
const itemContent2 = 'Something else';

describe('Publication component', () => {
  beforeEach(() => {
    rendered = render(
      <Router>
        <DisplayMenu data={displayMenuData} />
      </Router>
    );
  });

  test('should render', () => {
    const { asFragment } = rendered;
    expect(asFragment()).toMatchSnapshot();
  });

  //   test('should toggle item content', async () => {
  //     const { findAllBygetByText, getAllByText, queryByText } = rendered;
  //     console.log(getByText(displayMenuData[1].name).children);
  //     expect(getAllByText(itemContent1)).toBeTruthy();
  //     expect(queryByText(itemContent2)).toBeNull();
  //     fireEvent.click(getByText(displayMenuData[1].name));
  //     const newContent2 = await waitForElement(() => getByText(itemContent2));
  //     expect(newContent2).toBeTruthy();
  //   });

  test('should toggle main content', async () => {
    const { getByText, queryByText } = rendered;
    expect(getByText(content1)).toBeTruthy();
    expect(queryByText(content2)).toBeNull();
    fireEvent.click(getByText(displayMenuData[1].name));
    const newContent2 = await waitForElement(() => getByText(content2));
    expect(newContent2).toBeTruthy();
  });
});
