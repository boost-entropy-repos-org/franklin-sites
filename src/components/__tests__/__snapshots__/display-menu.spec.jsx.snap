// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display menu component should render 1`] = `
<DocumentFragment>
  <ul
    class="display-menu"
  >
    <ul
      class="no-bullet"
    >
      <li>
        <h5>
          <a
            aria-current="page"
            class="display-menu__item_title display-menu__item_title--active"
            href="///main"
          >
            <span
              class="display-menu__item_icon"
            >
              <test-file-stub />
            </span>
            Item 1
          </a>
        </h5>
        <div
          class="display-menu__item_content"
        >
          <ul
            class="no-bullet"
          >
            <li>
              Something
            </li>
            <li>
              Something
            </li>
            <li>
              Something
            </li>
            <li>
              Something
            </li>
            <li>
              Something
            </li>
          </ul>
        </div>
      </li>
      <li>
        <h5>
          <a
            class="display-menu__item_title"
            href="///item2"
          >
            <span
              class="display-menu__item_icon"
            />
            Item 2
          </a>
        </h5>
      </li>
      <li>
        <h5>
          <a
            class="display-menu__item_title"
            href="///item3"
          >
            <span
              class="display-menu__item_icon"
            />
            Item 3
          </a>
        </h5>
      </li>
    </ul>
  </ul>
</DocumentFragment>
`;
